sdk/internal/errorinfo/errorinfo_test.go

```diff
@@ -0,0 +1,39 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package errorinfo
+
+import (
+	"testing"
+)
+
+func TestNonRetriableError(t *testing.T) {
+	//Create sample error.
+	err := NonRetriableError(NewSampleError("Do Not Retry"))
+
+	// Check if the error is of type NonRetriableError
+	if _, ok := err.(*nonRetriableError); !ok {
+		t.Fatal("Expected error to be of type NonRetriableError")
+	}
+
+	// Check error message is correct
+	errMsg := "Do Not Retry"
+	if err.Error() != errMsg {
+		t.Fatalf("Expected error message to be '%q' but got '%q'.", errMsg, err.Error())
+	}
+}
+
+func NewSampleError(message string) error {
```

Comment:
If you want there's already a convenient "create an error based on a string" with `errors.New`.

(For future stuff, if you want to format the error there's `fmt.Errorf`.)

Reply:
(if the intention here was to make sure you could "unwrap" the inner type then you'll want to do an `errors.As()` in your test too.)